### PR TITLE
fixes vassals not seeing any huds and fixes bloodsucker huds

### PR DESCRIPTION
**Pull request by SmoSmoSmoSmok**
## About The Pull Request

 vassals can now see only their own bloodsucker master and only vassals that are on their team. BLOODSUCKERS can now only see their own vassals

## Changelog

:cl:
/:cl:

### CODE REVIEW

File: fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm [hunk 1]
```diff
@@ -65,6 +82,7 @@
 	owner.current.grant_all_languages(FALSE, FALSE, TRUE)
 	owner.current.grant_language(/datum/language/vampiric)
 	. = ..()
+	show_hud()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm:85]: can you call this update_hud_viewers() instead

File: fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm [hunk 2]
```diff
@@ -31,6 +31,23 @@
...
+	QDEL_NULL(team_hud_ref)
+
+	team_hud_ref = WEAKREF(target.add_alt_appearance(
+		/datum/atom_hud/alternate_appearance/basic/has_antagonist,
+		"antag_team_hud_[REF(src)]",
+		image(hud_icon, target, antag_hud_name),
+		antag_to_check || type,
+	))
+	var/list/mob/mob_list = list()
+	for( var/datum/antagonist/vassal/vassal as anything in master.vassals)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm:44]:
```suggestion
	for(var/datum/antagonist/vassal/vassal as anything in master.vassals)
```

File: fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm [hunk 3]
```diff
@@ -180,6 +199,19 @@
 	for(var/datum/action/cooldown/bloodsucker/power in powers)
 		power.level_current++
 
+/datum/antagonist/vassal/proc/show_hud()
+	var/datum/atom_hud/alternate_appearance/basic/has_antagonist/hud
+	for(var/datum/atom_hud/alternate_appearance/basic/has_antagonist/antag_hud as anything in GLOB.has_antagonist_huds)
+		if(antag_hud.target != owner.current)
+			continue
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm:206]: can we check ``team_hud_ref`` here instead? you gotta just resolve it.

File: fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm [hunk 4]
```diff
@@ -180,6 +199,19 @@
...
+		hud = antag_hud
+		break
+	if(!hud)
+		return
+
+	for( var/datum/antagonist/vassal/vassal as anything in master.vassals)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm:212]:
```suggestion
	for(var/datum/antagonist/vassal/vassal as anything in master.vassals)
```

File: fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm (hunk 3 above)
Comment (SmoSmoSmoSmok) [reply fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm]: > can we check `team_hud_ref` here instead? you gotta just resolve it.

how did i mis that

File: fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm [hunk 5]
```diff
@@ -29,7 +29,27 @@
...
+		/datum/atom_hud/alternate_appearance/basic/has_antagonist,
+		"antag_team_hud_[REF(src)]",
+		image(hud_icon, target, antag_hud_name),
+	))
+
+	var/datum/atom_hud/alternate_appearance/basic/has_antagonist/hud = team_hud_ref.resolve()
+
+	var/list/mob/mob_list = list()
+	for(var/datum/antagonist/vassal/vassal as anything in master.vassals)
+		mob_list += vassal.owner.current
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm:47]: can you also add the master to this list as well

File: fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm [hunk 6]
```diff
@@ -29,7 +29,27 @@
...
+
+	team_hud_ref = WEAKREF(target.add_alt_appearance(
+		/datum/atom_hud/alternate_appearance/basic/has_antagonist,
+		"antag_team_hud_[REF(src)]",
+		image(hud_icon, target, antag_hud_name),
+	))
+
+	var/datum/atom_hud/alternate_appearance/basic/has_antagonist/hud = team_hud_ref.resolve()
+
+	var/list/mob/mob_list = list()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm:45]:
```suggestion
	var/list/mob/living/mob_list = list()
```
and check for living too please, we don't want ghosts with antag huds

File: fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm [hunk 7]
```diff
@@ -29,7 +29,27 @@
...
+	for(var/datum/antagonist/vassal/vassal as anything in master.vassals)
+		mob_list += vassal.owner.current
+	for (var/datum/atom_hud/alternate_appearance/basic/has_antagonist/antag_hud as anything in GLOB.has_antagonist_huds)
+		if(!(antag_hud.target in mob_list) && antag_hud.target != master.owner.current)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/antagonists/vassal_datum.dm:49]: This list will become much more simple if you do the 2 above reviews

**Review by JohnFulpWillard — APPROVED** [review]

**Pull request merged**